src/box/txn_limbo.h

```diff
@@ -330,9 +335,29 @@ txn_limbo_rollback(struct txn_limbo *limbo)
 	latch_unlock(&limbo->promote_latch);
 }
 
-/** Apply a synchronous replication request after processing stage. */
+/**
+ * Prepare a limbo request for WAL write and commit. Similar to txn_stmt
+ * prepare.
+ */
+void
+txn_limbo_req_prepare(struct txn_limbo *limbo,
```

Comment:
Why did you decide not to merge `txn_limbo_req_prepare` into `txn_limbo_begin` ?

Reply:
My thinking was that firstly, not all synchro requests need the lock - `CONFIRM` and `ROLLBACK` seem to be working fine. Secondly, `txn_limbo_begin()` yields. After the fiber gets control, some conditions might change which will make the request invalid. For instance, 2 competing promotions arrive, one is written, the other one should fail after `txn_limbo_begin()` before it makes any changes. Third reason - I want it to look more like normal transactions: firstly you make `txn_begin()`, then inside of `txn_prepare()` you prepare individual statements. Same with the limbo and its requests. Except we don't have `txn_limbo_prepare()` - we prepare each request manually. At least so far.

Reply:
Ok, I see.